backend/test/handlers/SourceHandler.test.ts

```diff
@@ -25,8 +25,12 @@ jest.mock('fs-extra');
 describe('SourceHandler', () => {
     it('generates archive tree', () => {
         // @ts-ignore Passing mocked request
-        return browseArchiveTree({ params: {} }).then(archiveTree =>
-            expect(archiveTree?.children?.[0]?.children?.[0].key).toEqual('subdir/fileNameSpecial%3F%23Characters')
-        );
+        return browseArchiveTree({ params: {} }).then(archiveTree => {
+            expect(archiveTree).not.toBeNull();
+            expect(archiveTree && archiveTree?.children?.[0].isDir).toBeTruthy();
+            expect(
+                archiveTree && archiveTree?.children?.[0].isDir && archiveTree?.children?.[0]?.children?.[0].key
+            ).toEqual('subdir/fileNameSpecial%3F%23Characters');
```

Comment:
Wouldn't it be equal but shorter?
Wouldn't it be sufficient?
```suggestion
            expect(archiveTree?.children?.[0].isDir && archiveTree?.children?.[0].children[0].key
            ).toEqual('subdir/fileNameSpecial%3F%23Characters');
```
`archiveTree?.children?.[0]` could be also extracted to a variable

Reply:
I made it shorter:
[9c56437](https://github.com/cloudify-cosmo/cloudify-stage/pull/2467/commits/9c5643751387b22b55519abb4159e9a408def41c)

extracting to a variable didn't work:
<img width="813" alt="Screenshot 2023-03-21 at 12 12 10" src="https://user-images.githubusercontent.com/17104212/226576174-6d959041-4fe8-499c-ac5e-18f0a9af5c49.png">

Reply:
it can be changed to:
```typescript
const tree = archiveTree && archiveTree?.children?.[0];
expect(tree && tree.isDir && tree?.children?.[0].key).toEqual('subdir/fileNameSpecial%3F%23Characters');
```

Reply:
Since we are using `?.` this should work equally well I think:
```
const tree = archiveTree?.children?.[0];
```
and then we can use `?.` again for the first `tree` condition but drop it for the second one (I didn't validated it though):
```
expect(tree?.isDir && tree.children[0].key).toEqual('subdir/fileNameSpecial%3F%23Characters');
```

Reply:
I changed it: [352969e](https://github.com/cloudify-cosmo/cloudify-stage/pull/2467/commits/352969e5aabd0a4a76d0826cac078cb7782c1d40)